sdk/keyvault/keyvault-certificates/src/lro/core-lro-update.ts

```diff
@@ -175,7 +199,7 @@ export abstract class KVPoller<TState extends PollOperationState<TResult>, TResu
       const clearPollOncePromise = (): void => {
         this.pollOncePromise = undefined;
       };
-      this.pollOncePromise.then(clearPollOncePromise, clearPollOncePromise);
+      return this.pollOncePromise.then(clearPollOncePromise, clearPollOncePromise);
```

Comment:
Oof, this is a good one.

Reply:
I'm not sure if this change is safe actually.
With the current code, we always return `this.pollOncePromise`. With this code, we're now returning the promise that's chained off of `this.pollOncePromise` (in most cases given this.pollOncePromise will likely be undefined when poll is called).

I believe this has different behavior in the error case. In the original code, if `this.pollOnce` rejected with an error, the user could catch that in the code surrounding their `poll` call.

With this code change, the `then` callbacks will capture the error from `this.pollOncePromise` and doesn't rethrow it. Now if `this.pollOnce` rejects, the error is effectively swallowed.

At the very least, there is different behavior in the error case depending on whether `poll` while a pending poll is in progress or not.

Reply:
the alternative is to have an empty catch. I'll do that.

Reply:
I'll put it back and add an eslint line disabling the error only there.

Reply:
Undid the deed.